src/pytorch_metric_learning/utils/inference.py

```diff
@@ -93,11 +93,9 @@ def __init__(
         self.normalize_embeddings = normalize_embeddings
 
     def get_embeddings_from_tensor_or_dataset(self, inputs, batch_size):
-        if isinstance(inputs, list):
-            inputs = torch.stack(inputs)
-
+        inputs = self.process_if_list(inputs)
         embeddings = []
-        if torch.is_tensor(inputs):
+        if isinstance(inputs, (torch.Tensor, list)):
```

Comment:
Treat input as a batch of batches if input is a tensor **or list**, i.e. if the input has length 128, and the batch_size is 32, then the model will consume 4 batches.